algoliasearch/src/main/java/com/algolia/search/saas/AbstractClient.java

```diff
@@ -87,6 +88,24 @@ public int hashCode() {
         }
     }
 
+    static class HostStatus {
+        public boolean isUp = true;
+        public long lastTryTimestamp;
+
+        HostStatus() {
+            lastTryTimestamp = new Date().getTime();
+        }
+
+        HostStatus(boolean isUp) {
+            this.isUp = isUp;
+            lastTryTimestamp = new Date().getTime();
+        }
+
+        void updateState(boolean isUp) {
```

Comment:
Looks like dead code. `HostStatus` looks like an immutable class in spirit, anyway.